server/http/src/service.rs

```diff
@@ -189,18 +182,15 @@ where
         let body = hyper::body::to_bytes(request_body)
             .map_err(|_| GraphQLServerError::InternalError("Failed to read request body".into()))
             .await?;
-        let query = GraphQLRequest::new(body).compat().await;
+        let query = parse_graphql_request(&body);
 
         let result = match query {
             Ok(query) => service.graphql_runner.run_query(query, target).await,
             Err(GraphQLServerError::QueryError(e)) => QueryResult::from(e).into(),
             Err(e) => return Err(e),
         };
 
-        if let Some(id) = result.first().and_then(|res| res.deployment.clone()) {
```

Comment:
So it seems before we would only observe the metrics for known deployment ids, now if there's none the string `unknown` will be used, do we want to track metrics for those? Which scenarios would this value be `None`?

Reply:
That, and we also wouldn't record timing and counts for failed queries, something Martynas wanted to see for some investigation he was doing